compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java

```diff
@@ -134,28 +137,30 @@ public void visit(BLangTypeDefinition typeDefinition) {
         BLangType typeNode = typeDefinition.getTypeNode();
         if (typeDefinition.typeNode.getKind() == NodeKind.OBJECT_TYPE) {
             List<? extends VariableNode> fields = ((BLangObjectTypeNode) typeNode).getFields();
-            validateParameters(typeDefinition, fields, DiagnosticCode.UNDOCUMENTED_FIELD,
+            validateParameters(typeDefinition, fields, new LinkedList<>(), DiagnosticCode.UNDOCUMENTED_FIELD,
                     DiagnosticCode.NO_SUCH_DOCUMENTABLE_FIELD, DiagnosticCode.FIELD_ALREADY_DOCUMENTED);
             validateReturnParameter(typeDefinition, null, false);
 
             ((BLangObjectTypeNode) typeDefinition.getTypeNode()).getFunctions().forEach(this::analyzeNode);
         } else if (typeDefinition.typeNode.getKind() == NodeKind.RECORD_TYPE) {
             List<? extends VariableNode> fields = ((BLangRecordTypeNode) typeNode).getFields();
-            validateParameters(typeDefinition, fields, DiagnosticCode.UNDOCUMENTED_FIELD,
+            validateParameters(typeDefinition, fields, new LinkedList<>(), DiagnosticCode.UNDOCUMENTED_FIELD,
                     DiagnosticCode.NO_SUCH_DOCUMENTABLE_FIELD, DiagnosticCode.FIELD_ALREADY_DOCUMENTED);
             validateReturnParameter(typeDefinition, null, false);
         }
     }
 
     @Override
     public void visit(BLangResource resourceNode) {
-        validateParameters(resourceNode, resourceNode.getParameters(), DiagnosticCode.UNDOCUMENTED_PARAMETER,
-                DiagnosticCode.NO_SUCH_DOCUMENTABLE_PARAMETER, DiagnosticCode.PARAMETER_ALREADY_DOCUMENTED);
+        validateParameters(resourceNode, resourceNode.getParameters(), resourceNode.getDefaultableParameters(),
```

Comment:
should we consider rest parameters as well?

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/10298/commits/124252c01db7a58f2630339f2a2678a61bc27078.

Reply:
Ack.